ipysurfer/viz/brain.py

```diff
@@ -0,0 +1,541 @@
+        self._data['time_idx'] = time_idx
+        # data specific for a hemi
+        self._data[hemi + '_array'] = array
+
+        if time is not None and len(array.shape) == 2:
+            # we have scalar_data with time dimension
+            act_data = array[:, time_idx]
+        else:
+            # we have scalar data without time
+            act_data = array
+
+        if center is None:
+            if min is None:
+                min = array.min() if array.size > 0 else 0
+            if max is None:
+                max = array.max() if array.size > 0 else 1
+        else:
+            if min is None:
+                min = 0
+            if max is None:
+                max = np.abs(center - array).max() if array.size > 0 else 1
+        if mid is None:
+            mid = (min + max) / 2.
+        _check_limits(min, mid, max, extra='')
+        self._data['alpha'] = alpha
+        self._data['colormap'] = colormap
+        self._data['center'] = center
+        self._data['min'] = min
+        self._data['mid'] = mid
+        self._data['max'] = max
+
+        lut = self.update_lut()
+
+        # Create smoothing matrix if necessary
+        if len(act_data) < self.geo[hemi].x.shape[0]:
+            if vertices is None:
+                raise ValueError('len(data) < nvtx (%s < %s): the vertices '
+                                 'parameter must not be None'
+                                 % (len(act_data), self.geo[hemi].x.shape[0]))
+            adj_mat = _mesh_edges(self.geo[hemi].faces)
+            smooth_mat = _smoothing_matrix(vertices,
+                                           adj_mat,
+                                           smoothing_steps)
+            act_data = smooth_mat.dot(act_data)
+            self._data[hemi + '_smooth_mat'] = smooth_mat
+        else:
+            smooth_mat = None
+
+        # data mapping into [0, 1] interval
+        dt_max = max
+        dt_min = min if center is None else -1 * max
+        k = 1 / (dt_max - dt_min)
+        b = 1 - k * dt_max
+        act_data = k * act_data + b
+        act_data = np.clip(act_data, 0, 1)
+
+        act_color = lut(act_data)
+
+        self._data['k'] = k
+        self._data['b'] = b
+
+        for ri, v in enumerate(self._views):
+            if self._hemi != 'split':
+                ci = 0
+            else:
+                ci = 0 if hemi == 'lh' else 1
+            ipv.pylab.current.figure = self._figures[ri][ci]
+            hemi_overlay = self._plot_hemi_overlay(self.geo[hemi].coords,
+                                                   self.geo[hemi].faces,
+                                                   act_color)
+            self._overlays[hemi + '_' + v] = hemi_overlay
+
+        if colorbar and not self._colorbar_added:
+            ColorBar(self)
+            self._colorbar_added = True
+
+    def show(self):
+        u"""Display widget."""
+        ipv.show()
+
+    def update_lut(self, min=None, mid=None, max=None):
```

Comment:
what does "lut" stand for? if the method updates teh colormap, let's just call it "update_colors" or something

Reply:
As I know, "lut" stands for "Look-up table". I have borrowed naming from `PySurfer`. I will change it to something more obvious.

Reply:
if this is just copy-pasted from ipysurfer then I think it's fine...